flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/AlterSchemaConverter.java

```diff
@@ -110,6 +113,7 @@ public Operation convertAlterSchema(
         SchemaConverter converter = createSchemaConverter(alterTableSchema, oldTable);
         converter.updateColumn(alterTableSchema.getColumnPositions().getList());
         alterTableSchema.getWatermark().ifPresent(converter::updateWatermark);
+        alterTableSchema.getDistribution().ifPresent(converter::updateDistribution);
```

Comment:
I thought the main change in this class is only to update the distribution in case of a column name change similar to  `buildUpdatedPrimaryKey`.

Reply:
I _think_ we have to pass the distribution around in order to create the new table objects.  I may be missing some alternative here.